Rename ProjectSpec -> ProjectSpec.Project

jerrymarino opened the pull request:
I've started using the XcodeGen data structures in a large project, and
its been awesome so far!

The problem I'm running into is I can't import types from the
`ProjectSpec` module. This caused by a naming collision between the struct
`ProjectSpec` and the module `ProjectSpec`.

To remedy the issue, this patch renames the struct to `Project`. This is
an overall improvement, since all of the types are now nicely namespaced
and can be imported as expected.

i.e.
```
import ProjectSpec

let projectSpec: ProjectSpec.Project
let targetSpec: ProjectSpec.Target
```

This makes it easier to differentiate in the code when we are dealing
with types of similar names, like `PBXProject`.

Any feedback is welcome 👍

yonaskolb commented on Sources/XcodeGenKit/SettingsBuilder.swift:
```diff
@@ -112,7 +112,7 @@ extension ProjectSpec {
         if let configSettings = loadConfigFileBuildSettings(path: configPath) {
             for key in configSettings.keys {
-                // FIXME: Catch platform specifier. e.g. LD_RUNPATH_SEARCH_PATHS[sdk=iphone*]
+                // FIXME: Catch platform projectifier. e.g. LD_RUNPATH_SEARCH_PATHS[sdk=iphone*]
```
Search and replace typo

jerrymarino replied:
Good catch!

yonaskolb commented on Tests/XcodeGenKitTests/ProjectGeneratorTests.swift:
```diff
@@ -110,7 +110,7 @@ func projectGeneratorTests() {
             $0.it("generates configs") {
-                let spec = ProjectSpec(
+                let spec = ProjectSpec.Project(
                     basePath: "",
```
It seems everything in Tests has a different renaming applied. Does leaving off the `ProjectSpec` module cause naming conflicts?

jerrymarino replied:
Hmm - we shouldn't have a naming conflict, will update to match the rest of the renames!

jerrymarino commented:
@yonaskolb thanks again for the review, and sorry for the latency on this. I've rebased this on master and updated the test renaming to match the rest 👍

yonaskolb commented:
Thanks @jerrymarino! I've added your changes into here https://github.com/yonaskolb/XcodeGen/pull/281
